### DISCUSSION TITLE
Use Mobile device name in Sync View

### DISCUSSION THREAD

**Discussion by arky** [digidem/mapeo-mobile#768] [Ideas]
I think it is a good idea to get and use the device name of mobile instead of the current Android and randomized string.  Perhaps using react-native-device info's [getDeviceName()](https://github.com/react-native-community/react-native-device-info#getdevicename).  

This was the feedback from participants of IP workshop in Chiang Mai. /CC @prustar

https://github.com/digidem/mapeo-mobile/blob/9681427037a637ef318b95fac6632ff8c9208e87/src/frontend/screens/SyncModal/index.js#L38-L43

**Comment by gmaclennan** [comment]
Hi @arky, one of the challenges with this is that there is not a clear concept of "device name" on Android. The value returned from this method is the name use by Android for bluetooth discovery and it is quite hidden in the settings. Most phones use their model name as the default for this name e.g. "Motorola G3". We actually used this function initially, but because we tend to deploy Mapeo with multiples of the same phone, the sync screen just shows a list of identical names. It's possible to train users to set their bluetooth device name, but it's confusing if they forget to do it.

Our proposed solution to this is to integrate a "device naming" step in the initial setup of Mapeo Mobile e.g. when you first install it, it asks you to name your device (this would be a name for the device only used by Mapeo). We also want to use this same name in order to track sync history for peers within a project.
